snuba/admin/views.py

```diff
@@ -111,6 +112,42 @@ def migrations_groups_list(group: str) -> Response:
     return make_response(jsonify({"error": "Invalid group"}), 400)
 
 
+class MigrationActionConverter(BaseConverter):
+    regex = r"(?:run|reverse)"
+
+
+application.url_map.converters["migration_action"] = MigrationActionConverter
+
+
+@application.route(
+    "/migrations/<group>/<migration_action:action>/<migration_id>",
+    methods=["POST"],
+)
+def run_or_reverse_migration(group: str, action: str, migration_id: str) -> Response:
+    if group not in settings.ADMIN_ALLOWED_MIGRATION_GROUPS:
+        return make_response(jsonify({"error": "Group not allowed"}), 400)
+
+    runner = Runner()
+    migration_group = MigrationGroup(group)
+    migration_key = MigrationKey(migration_group, migration_id)
+
+    def str_to_bool(s: str) -> bool:
+        return s.strip().lower() in ("yes", "true", "t", "1")
+
+    force = request.args.get("force", False, type=str_to_bool)
+    fake = request.args.get("fake", False, type=str_to_bool)
+
+    try:
+        if action == "run":
+            runner.run_migration(migration_key, force=force, fake=fake)
+        else:
+            runner.reverse_migration(migration_key, force=force, fake=fake)
+    except Exception as e:
+        return make_response(jsonify({"error": repr(e)}), 500)
```

Comment:
You should also do `logger.error(...` here so that the error gets sent to Sentry.

Reply:
It's best practice to only expose an error message in the API, and not the underlying Python logic.

Reply:
fixed